entities/src/main/java/org/odk/collect/entities/LocalEntityUseCases.kt

```diff
@@ -62,11 +62,14 @@ object LocalEntityUseCases {
     fun updateLocalEntitiesFromServer(
         list: String,
         serverList: File,
-        entitiesRepository: EntitiesRepository
+        entitiesRepository: EntitiesRepository,
+        entitySource: EntitySource,
+        serverListHash: String,
+        integrityUrl: String?
     ) {
-        val listHash = getListHash(serverList)
-        val existingListVersion = entitiesRepository.getListHash(list)
-        if (listHash == existingListVersion) {
+        val newListHash = "server:$serverListHash"
```

Comment:
We need to use the server hash here instead of the file hash to prevent problems coming up with a "new" entity list having the same contents as a previous one due to deletions.